### PR TITLE
use sample manager in allele freq column

### PR DESCRIPTION
**Pull request by adamabeshouse**
# Checks
- [x] Follows [7 rules of great commit messages](http://chris.beams.io/posts/git-commit/). For most PRs a single commit should suffice, in some cases multiple topical commits can be useful. During review it is ok to see tiny commits (e.g. Fix reviewer comments), but right before the code gets merged to master or rc branch, any such commits should be squashed since they are useless to the other developers. Definitely avoid [merge commits, use rebase instead.](http://nathanleclaire.com/blog/2014/09/14/dont-be-scared-of-git-rebase/)
- [x] Follows the [Airbnb React/JSX Style guide](https://github.com/airbnb/javascript/tree/master/react).
- [x] Make sure your commit messages end with a Signed-off-by string (this line
  can be automatically added by git if you run the `git-commit` command with
  the `-s` option)

### CODE REVIEW

File: src/pages/patientView/sampleManager.tsx [hunk 1]
```diff
@@ -18,10 +18,12 @@ export function placeArrow(tooltipEl: any) {
 class SampleManager {
 
     sampleIndex: { [s:string]:number };
+    sampleIdsInOrder: string[];
 
     constructor(public samples: Array<ClinicalDataBySampleId>){
 
         this.sampleIndex = {};
+        this.sampleIdsInOrder = samples.map((sample:ClinicalDataBySampleId) => sample.id);
```
Comment (alisman) [inline src/pages/patientView/sampleManager.tsx]: troubles me slightly that we now have two orderings:  1. the order of samples in samples collection and two the order in this array.  Is there a reason not to just iterate through samples collection in your calling code, using sample.id?  (I admit's it's a tiny bit weird that the samples collection is filled with things of type ClinicalDataBySampleId ... )

File: src/pages/patientView/mutation/column/AlleleFreqColumnFormatter.tsx [hunk 2]
```diff
@@ -14,56 +15,55 @@ export default class AlleleFreqColumnFormatter {
...
+                    placement="left"
+                    overlay={tooltipLines}
+                    arrowContent={<div className="rc-tooltip-arrow-inner"/>}
                 >
-                    {bars}
-                </svg>
-            </Tooltip>
+
+                    <svg
+                        width={sampleOrder.length*AlleleFreqColumnFormatter.barWidth + (sampleOrder.length-1)*AlleleFreqColumnFormatter.barSpacing}
```
Comment (alisman) [inline src/pages/patientView/mutation/column/AlleleFreqColumnFormatter.tsx]: maybe put this in a testable method

File: src/pages/patientView/sampleManager.tsx (hunk 1 above)
Comment (adamabeshouse) [inline src/pages/patientView/sampleManager.tsx]: thats the issue. Maybe this is a good candidate for reselect?

File: src/pages/patientView/mutation/column/AlleleFreqColumnFormatter.tsx (hunk 2 above)
Comment (adamabeshouse) [inline src/pages/patientView/mutation/column/AlleleFreqColumnFormatter.tsx]: which part?

Comment (alisman) [inline src/pages/patientView/mutation/column/AlleleFreqColumnFormatter.tsx]: the width calculation

Comment (adamabeshouse) [inline src/pages/patientView/mutation/column/AlleleFreqColumnFormatter.tsx]: done

**Comment by adamabeshouse** [comment]
[Use sample manager for allele frequency column](https://trello.com/c/q0Xuwxd6/64-use-sample-manager-for-allele-frequency-column)

**Pull request merged**